app/gui/src/model/project/synchronized.rs

```diff
@@ -708,7 +715,17 @@ impl model::project::API for Project {
             let project_manager = self.project_manager.as_ref().ok_or(ProjectManagerUnavailable)?;
             let project_id = self.properties.borrow().id;
             let project_name = ProjectName::new_unchecked(name);
-            project_manager.rename_project(&project_id, &project_name).await?;
+            project_manager.rename_project(&project_id, &project_name).await.map_err(|error| {
+                match error {
+                    RpcError::RemoteError(cause) =>
+                        if cause.code == 4001 {
+                            failure::Error::from(ProjectNameInvalid { cause: cause.message })
+                        } else {
+                            RpcError::RemoteError(cause).into()
+                        },
+                    error => error.into(),
+                }
```

Comment:
This logic would be simpler with a match guard.

Reply:
Oh yeah, good catch!